src/Juvix/Compiler/Concrete/Translation/FromSource.hs

```diff
@@ -564,11 +564,20 @@ parsePatternAtoms = do
 
 functionClause :: Members '[InfoTableBuilder, JudocStash, NameIdGen] r => Symbol -> ParsecS r (FunctionClause 'Parsed)
 functionClause _clauseOwnerFunction = do
-  _clausePatterns <- P.many patternAtom
-  kw kwAssign
+  _clausePatterns <- functionClausePatterns
   _clauseBody <- parseExpressionAtoms
   return FunctionClause {..}
 
+functionClausePatterns :: forall r. Members '[InfoTableBuilder, JudocStash, NameIdGen] r => ParsecS r [PatternAtom 'Parsed]
+functionClausePatterns =
+  P.manyTill patternAtom (kw kwAssign <|> wrongEq)
+  where
+    wrongEq :: ParsecS r ()
+    wrongEq = do
+      P.chunk "="
+      off <- P.getOffset
+      parseFailure off "expected \":=\" instead of \"=\""
```

Comment:
this error message does not seem to surface. 
I am getting this:
![image](https://user-images.githubusercontent.com/5511599/211681127-497b8897-187d-4174-9dd2-e26ce4abe3b7.png)


To fix it you can do this:
```diff
-  P.manyTill patternAtom (kw kwAssign <|> wrongEq)
+  P.manyTill patternAtom (wrongEq <|> kw kwAssign)
   where
     wrongEq :: ParsecS r ()
     wrongEq = do
-      P.chunk "="
-      off <- P.getOffset
-      parseFailure off "expected \":=\" instead of \"=\""
+      void (P.single '=')
+      P.failure (Just (P.Tokens (pure '='))) (Set.singleton (P.Tokens (':' :| "=")))
```

Reply:
you'll need `import Data.Set qualified as Set`

Reply:
No longer relevant since the fix above.